yt_dlp/extractor/youtube/_base.py

```diff
@@ -595,6 +601,31 @@ def _get_sid_authorization_header(self, origin='https://www.youtube.com', user_s
 
         return ' '.join(authorizations)
 
+    @property
+    def is_authenticated(self):
+        return self._has_auth_cookies
+
+    @property
+    def _has_auth_cookies(self):
+        yt_sapisid, yt_1psapisid, yt_3psapisid = self._get_sid_cookies()
+        # YouTube doesn't appear to clear 3PSAPISID when rotating cookies (as of 2025-04-26)
+        # But LOGIN_INFO is cleared and should exist if logged in
+        has_login_info = 'LOGIN_INFO' in self._youtube_cookies
+        return has_login_info and (yt_sapisid or yt_1psapisid or yt_3psapisid)
+
+    def _request_webpage(self, *args, **kwargs):
+        response = super()._request_webpage(*args, **kwargs)
+
+        # Check that we are still logged-in and cookies have not rotated after every request
+        if self._had_cookie_auth and not self._has_auth_cookies:
+            self.report_warning(
+                'The provided YouTube account cookies are no longer valid. '
+                'They have likely been rotated in the browser as a security measure. '
+                f'For tips on how to effectively export YouTube cookies, refer to  {self._COOKIE_HOWTO_WIKI_URL}  .',
+                only_once=False)
```

Comment:
did you want to be deliberate with `only_once=False`? or could we do this?
```suggestion
                f'For tips on how to effectively export YouTube cookies, refer to  {self._COOKIE_HOWTO_WIKI_URL}')
```

Reply:
Yeah I wanted it to be deliberate